common/src/main/java/tachyon/security/UserInformation.java

```diff
@@ -114,10 +146,89 @@ static String getOsPrincipalClassName() {
 
     @Override
     public AppConfigurationEntry[] getAppConfigurationEntry(String appName) {
-      // TODO: switch branch based on appName. (Simple, Kerberos, etc)
-      return SIMPLE;
+      if (appName.equalsIgnoreCase(AuthenticationFactory.AuthTypes.SIMPLE.getAuthName())) {
+        return SIMPLE;
+      } else if (appName.equalsIgnoreCase(AuthenticationFactory.AuthTypes.KERBEROS.getAuthName())) {
+        // TODO: return KERBEROS;
+        throw new UnsupportedOperationException("Kerberos is not supported currently.");
+      }
+      return null;
+    }
+  }
+
+  /**
+   * Set the static configuration and initialize based on it
+   * @param conf the configuration
+   */
+  public static void setTachyonConf(TachyonConf conf) {
+    initialize(conf);
+  }
+
+  /**
+   * Check whether fields are initialized based on the configuration.
+   */
+  private static void ensureInitialized() {
+    if (sConf == null) {
+      initialize(new TachyonConf());
+    }
+  }
+
+  /**
+   * Initialize with conf
+   * @param conf the configuration
+   */
+  private static void initialize(TachyonConf conf) {
```

Comment:
This is confusing. If static methods are used, they meant to do something without maintaining state, which looks to me not the case here --- `sConf` seems a state set externally. If you want to maintain states like a `TachyonConf` or a `AuthTypes`, probably reconsider how to structure this class.